### PR TITLE
Authorization with new server fs

### PR DESCRIPTION
**Pull request by AntonBurduzha**
#5  and #23 ;

### CODE REVIEW

**Comment by AntonBurduzha** [comment]
there are some mistakes in function names and crud; should be fixed in next commit;

**Comment by AntonBurduzha** [comment]
next commit should contain server sign-up processing

File: .eslintrc [hunk 1]
```diff
@@ -5,7 +5,10 @@
   },
   "extends": ["airbnb"],
   "rules": {
-    "no-console": 0
+    "no-console": 0,
+    "no-trailing-spaces": 0,
```
Comment (dimagreat) [inline .eslintrc]: We already talked about editing .eslintrc,
please move your lint problems directly to files
and leave .eslintrc unchanged

File: app/user/sign-in/sign-in.controller.js [hunk 2]
```diff
@@ -1,14 +1,31 @@
-class SignInController {
-  constructor(UserService, $log, $translate) {
-    UserService.getGreetings()
-      .then((greetings) => {
-        $log.info(greetings);
+function SignInController(UserService, $log, $translate) {
+  const vm = this;
+  vm.language = 'en';
+  vm.password = '';
+  vm.login = '';
```
Comment (dimagreat) [inline app/user/sign-in/sign-in.controller.js]: Let's use single object for these values.

File: app/user/sign-in/sign-in.controller.js [hunk 3]
```diff
@@ -1,14 +1,31 @@
...
+
+  vm.changeLanguage = changeLanguage;
```
Comment (dimagreat) [inline app/user/sign-in/sign-in.controller.js]: What advantage does this syntax have?

File: app/user/user.service.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
+function UserService($http, $q) {
+  function postAuthData(userAuthData) {
+    const authData = JSON.stringify(userAuthData);
+
+    return $http.post('/oauth/token', authData)
+      .then(response => response.data)
+      .catch(response => $q.reject(response.data.error));
+  }
+
+  function postNewUserData(newUserData) {
```
Comment (dimagreat) [inline app/user/user.service.js]: Let's use more classic naming for methods;
createUser, loginUser
and dto as arguments

File: backend/services/auth.service.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
+const jwt = require('jsonwebtoken');
+const db = require('../db/db.config');
+require('dotenv').config();
```
Comment (dimagreat) [inline backend/services/auth.service.js]: Let's use this in app.js to avoid duplicate call

File: webpack.config.js [hunk 6]
```diff
@@ -4,13 +4,13 @@ const autoprefixer = require('autoprefixer');
...
 const defineWebpackPlugins = (prod) => {
   const plugins = [
-    new webpack.LoaderOptionsPlugin({
-      options: {
-        eslint: {
-          configFile: './.eslintrc',
-        },
-      },
-    }),
+    // new webpack.LoaderOptionsPlugin({
```
Comment (dimagreat) [inline webpack.config.js]: Please revert this changes

File: backend/services/picker.service.js [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+          if (err) {
+            reject({ error: 'An error has occurred' });
+          } else {
+            resolve(quizzesList);
+          }
+        });
+    });
+  }
+
+  function getSpecificQuiz(req) {
```
Comment (dimagreat) [inline backend/services/picker.service.js]: Why do you change this name?
You think `getSpecificQuiz` is more obvious for route /picker/:id then `getQuizById` ?

File: backend/services/picker.service.js [hunk 8]
```diff
@@ -0,0 +1,36 @@
+const db = require('../db/db.config');
+const ObjectID = require('mongodb').ObjectID;
+
+function PickerService() {
+  this.getQuizzesList = getQuizzesList;
```
Comment (dimagreat) [inline backend/services/picker.service.js]: Please refactor this to object, like angular factory

File: backend/controllers/picker.controller.js [hunk 9]
```diff
@@ -0,0 +1,29 @@
+const express = require('express');
+const PickerService = require('../services/picker.service');
+
+const router = express.Router();
+const pickerService = new PickerService();
+router.get('/picker', getQuizzesList);
+router.get('/picker/:id', getSpecificQuiz);
+
+function getQuizzesList(req, res) {
```
Comment (dimagreat) [inline backend/controllers/picker.controller.js]: It is identical function with service handle, please create single one

Comment (dimagreat) [inline backend/controllers/picker.controller.js]: Actually, all your router functions are identical

File: app/user/sign-in/sign-in.controller.js [hunk 10]
```diff
@@ -1,14 +1,31 @@
...
+
+  /*
+  *   Function for checking inputed user data and getting access token
+  */
+  function onLogin() {
+    if (vm.login && vm.password) {
+      const userAuthData = { login: vm.login, password: vm.password };
+
+      UserService.postAuthData(userAuthData).then((result) => {
+        console.log(result);
```
Comment (dimagreat) [inline app/user/sign-in/sign-in.controller.js]: Is is dev stuff here? How do we get picker screen after this log?

File: app/user/sign-up/sign-up.controller.js [hunk 11]
```diff
@@ -1,11 +1,57 @@
...
+        console.log(result);
+      });
+    }
+  }
+
+  /*
+  *   Helper function for checking empty form fields. Return false if
+  *   field is empty and name not equal adminPromoCode
+  */
+  function checkEmptyFields() {
```
Comment (dimagreat) [inline app/user/sign-up/sign-up.controller.js]: You can avoid using this function with using angular.FormController

File: app/user/sign-up/sign-up.controller.js [hunk 12]
```diff
@@ -1,11 +1,57 @@
...
+        return false;
+      }
+    }
+    return true;
+  }
+
+  /*
+  *   Helper function for checking password and confirm password equality
+  */
+  function checkPasswordConfirmation() {
```
Comment (dimagreat) [inline app/user/sign-up/sign-up.controller.js]: Same for FormController

File: app/user/user.service.js [hunk 13]
```diff
@@ -0,0 +1,24 @@
+function UserService($http, $q) {
+  function postAuthData(userAuthData) {
+    const authData = JSON.stringify(userAuthData);
+
+    return $http.post('/oauth/token', authData)
```
Comment (dimagreat) [inline app/user/user.service.js]: Why don't use route 'api/login' ?

File: backend/app.js [hunk 14]
```diff
@@ -16,3 +26,19 @@ db.connect((err) => {
     });
   }
 });
+
+const http = require('http').Server(app);
```
Comment (dimagreat) [inline backend/app.js]: What problem with using `app.listen`?

File: backend/auth.strategy.js [hunk 15]
```diff
@@ -0,0 +1,22 @@
+require('dotenv').config();
+const passportJWT = require('passport-jwt');
+const db = require('./db/db.config');
+
+const ExtractJwt = passportJWT.ExtractJwt;
+const JwtStrategy = passportJWT.Strategy;
+const jwtOptions = {
+  jwtFromRequest: ExtractJwt.fromAuthHeader(),
+  secretOrKey: process.env.CLIENT_SECRET,
```
Comment (dimagreat) [inline backend/auth.strategy.js]: Where I can find this `sercret`?

File: backend/auth.strategy.js [hunk 16]
```diff
@@ -0,0 +1,22 @@
...
+};
+
+const strategy = new JwtStrategy(jwtOptions, (jwtPayload, next) => {
+  db.get().collection('users').findOne({ login: jwtPayload.login }, (err, user) => {
```
Comment (dimagreat) [inline backend/auth.strategy.js]: It's look like method of authService

File: app/user/sign-in/sign-in.controller.js (hunk 3 above)
Comment (AntonBurduzha) [inline app/user/sign-in/sign-in.controller.js]: if controller will grow, this syntax give us smart navigation (ctrl + left mouse key)

File: app/user/sign-in/sign-in.controller.js (hunk 10 above)
Comment (AntonBurduzha) [inline app/user/sign-in/sign-in.controller.js]: this is temporary

File: app/user/user.service.js (hunk 13 above)
Comment (AntonBurduzha) [inline app/user/user.service.js]: because same endpoints i'm using at work projects.

Comment (dimagreat) [inline app/user/user.service.js]: It's not a good argument :)
Let's use our convetion

File: app/user/sign-in/sign-in.controller.js (hunk 3 above)
Comment (dimagreat) [inline app/user/sign-in/sign-in.controller.js]: I don't think it's necessary because we willn't use controllers more than 100 lines

File: .eslintrc (hunk 1 above)
Comment (AntonBurduzha) [inline .eslintrc]: +

File: app/user/sign-in/sign-in.controller.js (hunk 2 above)
Comment (AntonBurduzha) [inline app/user/sign-in/sign-in.controller.js]: completed

File: app/user/sign-in/sign-in.controller.js (hunk 3 above)
Comment (AntonBurduzha) [inline app/user/sign-in/sign-in.controller.js]: completed

File: app/user/sign-in/sign-in.controller.js (hunk 10 above)
Comment (AntonBurduzha) [inline app/user/sign-in/sign-in.controller.js]: completed

File: app/user/user.service.js (hunk 13 above)
Comment (AntonBurduzha) [inline app/user/user.service.js]: completed

File: app/user/user.service.js (hunk 4 above)
Comment (AntonBurduzha) [inline app/user/user.service.js]: completed

File: backend/services/auth.service.js (hunk 5 above)
Comment (AntonBurduzha) [inline backend/services/auth.service.js]: completed

File: webpack.config.js (hunk 6 above)
Comment (AntonBurduzha) [inline webpack.config.js]: completed

File: backend/services/picker.service.js (hunk 7 above)
Comment (AntonBurduzha) [inline backend/services/picker.service.js]: completed

File: backend/app.js (hunk 14 above)
Comment (AntonBurduzha) [inline backend/app.js]: completed

File: backend/auth.strategy.js (hunk 15 above)
Comment (AntonBurduzha) [inline backend/auth.strategy.js]: completed

File: backend/auth.strategy.js (hunk 16 above)
Comment (AntonBurduzha) [inline backend/auth.strategy.js]: completed

File: backend/services/picker.service.js (hunk 8 above)
Comment (AntonBurduzha) [inline backend/services/picker.service.js]: completed

File: app/user/sign-in/sign-in.controller.js [hunk 17]
```diff
@@ -11,11 +13,14 @@ function SignInController(UserService, $log, $translate) {
   *   Function for checking inputed user data and getting access token
   */
   function onLogin() {
-    if (vm.login && vm.password) {
-      const userAuthData = { login: vm.login, password: vm.password };
-
-      UserService.postAuthData(userAuthData).then((result) => {
-        console.log(result);
+    if (vm.signInForm.$valid) {
+      console.log(1);
```
Comment (dimagreat) [inline app/user/sign-in/sign-in.controller.js]: )

Comment (AntonBurduzha) [inline app/user/sign-in/sign-in.controller.js]: this is for debugging)))

Comment (AntonBurduzha) [inline app/user/sign-in/sign-in.controller.js]: resolved
